devito/arguments.py

```diff
@@ -52,37 +53,38 @@ def as_symbol(self):
         return Symbol(self.name)
 
     @property
-    def ready(self):
-        return self._value is not None
-
-    @property
     def dtype(self):
         return self.provider.dtype
 
     def reset(self):
-        self._value = self._default_value
+        self._value = self.default_value
 
     @abc.abstractproperty
     def verify(self, kwargs):
         return
 
+    def __repr__(self):
+        return self.name
+
 
 class ScalarArgument(Argument):
 
     """ Class representing scalar arguments that a kernel might expect.
         Most commonly used to pass dimension sizes
+        enforce determines whether any reduction will be performed or not.
+        i.e. if it is a user-provided value, use it directly.
     """
 
     is_ScalarArgument = True
 
-    def __init__(self, name, provider, reducer, default_value=None):
+    def __init__(self, name, provider, reducer=lambda old, new: new, default_value=None):
         super(ScalarArgument, self).__init__(name, provider, default_value)
         self.reducer = reducer
 
-    def verify(self, value):
+    def verify(self, value, engine, enforce=False):
```

Comment:
unused ``engine``

Reply:
For consistency, all verify methods accept engine.